shared/fs/send-to-chat/attachment/index.tsx

```diff
@@ -22,20 +22,36 @@ type Props = {
 
 const useConversationList = () => {
   const sendAttachmentToChat = Container.useSelector(state => state.fs.sendAttachmentToChat)
+  const {filter} = sendAttachmentToChat
   const dispatch = Container.useDispatch()
   const onSelect = (convID: ChatTypes.ConversationIDKey) => {
     dispatch(FsGen.createSetSendAttachmentToChatConvID({convID}))
   }
   const onDone = React.useCallback(() => dispatch(Chat2Gen.createToggleInboxSearch({enabled: false})), [
     dispatch,
   ])
-  const onSetFilter = (filter: string) => {
-    dispatch(FsGen.createSetSendAttachmentToChatFilter({filter}))
-    dispatch(Chat2Gen.createInboxSearch({query: new HiddenString(filter)}))
-  }
+  const onSetFilter = React.useCallback(
+    (filter: string) => {
+      dispatch(FsGen.createSetSendAttachmentToChatFilter({filter}))
+      dispatch(Chat2Gen.createInboxSearch({query: new HiddenString(filter)}))
+    },
+    [dispatch]
+  )
   React.useEffect(() => onDone, [onDone])
+
+  const layout = Container.useSelector(state => state.chat2.inboxLayout)
+
+  const filterEmpty = !filter
+
+  // force reload
+  React.useEffect(() => {
+    if (filterEmpty && layout) {
+      onSetFilter('')
+    }
+  }, [layout, onSetFilter, filterEmpty])
```

Comment:
Could you elaborate why it reloads on `layout` change? (perhaps in comment as well)

Reply:
this is going to change since we're moving to an rpc but, it depends on layout since we load the inboxlayout when this is spawned